src/chat.rs

```diff
@@ -1744,12 +1744,34 @@ pub async fn get_chat_msgs(
     }
 }
 
+pub(crate) async fn marknoticed_chat_if_older_than(
+    context: &Context,
+    chat_id: ChatId,
+    timestamp: i64,
+) -> Result<(), Error> {
+    if let Some(chat_timestamp) = context
+        .sql
+        .query_get_value(
+            context,
+            "SELECT MAX(timestamp) FROM msgs WHERE chat_id=?",
+            paramsv![chat_id],
+        )
+        .await
+    {
+        if timestamp > chat_timestamp {
```

Comment:
`chat_timestamp < timestamp` is maybe more readable, however, rust fails on type-inferring then and we would need to add some other clutter somewhere.